newrelic/core/attribute.py

```diff
@@ -282,14 +279,40 @@ def process_user_attribute(name, value, max_length=MAX_ATTRIBUTE_LENGTH, ending=
 
 
 def sanitize(value):
+    """
+    Return value unchanged, if it's a valid type that is supported by
+    Insights. Otherwise, convert value to a string.
 
-    # Return value unchanged, if it's a valid type that is supported by
-    # Insights. Otherwise, convert value to a string.
-    #
-    # Raise CastingFailureException, if str(value) somehow fails.
+    Raise CastingFailureException, if str(value) somehow fails.
+    """
 
     valid_value_types = (six.text_type, six.binary_type, bool, float, six.integer_types)
 
+    # When working with numpy, note that numpy has its own `int`s, `str`s,
+    # et cetera. This is fine for computational purposes, but when it
+    # comes to displaying this data in Insights/Nerdgraph, `numpy.str_`
+    # is not supported, unlike `str`.  `numpy.str_` is a subclass of string
+    # which `isinstance` captures.  Unfortunately, in this case, this
+    # is not the desired behavior.  `type()` needs to be explicitly used
+    # in this case and not `isinstance()` to ensure that anything other
+    # than what is in `valid_value_types` is converted to a `str`.
+    #
+    # In other words:
+    #   isinstance([var with type <class 'numpy.str_'>], <class 'str'>) == True
+    #   type([var with type <class 'numpy.str_'>]) != <class 'str'>
+    #
+
+    # This is to capture any potential numpy types that could
+    # otherwise correspond to a standard python type and
+    # converts them to the appropriate standard python type.
+    # If this is not the case, the function will attempt to
+    # convert the value into a string type. (e.g. UUID types)
+    if not type(value) in valid_value_types:
```

Comment:
I'm concerned about the performance impact this might have on the rest of the agent code for customers that add a lot of custom attributes. I wonder if it might be better to do this just inside the mlops code when setting the attribute?